app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeavePresenter.java

```diff
@@ -73,83 +73,67 @@ public boolean validateDataInput(Request request) {
             case RequestLeaveViewModel.LeaveType.IN_LATE_WOMAN_M:
-                if (validateCheckinTime(request.getCheckinTime()) & validateProjectName(
-                        request.getProject())) {
+                if (validateCheckinTime(request.getCheckinTime())) {
                     return true;
                 }
                 break;
             case RequestLeaveViewModel.LeaveType.LEAVE_EARLY_WOMAN_A:
             case RequestLeaveViewModel.LeaveType.LEAVE_EARLY_WOMAN_M:
-                if (validateCheckoutTime(request.getCheckoutTime()) & validateProjectName(
-                        request.getProject())) {
+                if (validateCheckoutTime(request.getCheckoutTime())) {
                     return true;
                 }
                 break;
             case RequestLeaveViewModel.LeaveType.FORGOT_CARD_ALL_DAY:
             case RequestLeaveViewModel.LeaveType.FORGOT_CHECK_ALL_DAY:
                 if (validateCheckinTime(request.getCheckinTime()) & validateCheckoutTime(
-                        request.getCheckoutTime()) & validateProjectName(request.getProject())) {
+                        request.getCheckoutTime())) {
                     return true;
                 }
 
                 break;
             case RequestLeaveViewModel.LeaveType.FORGOT_CARD_IN:
             case RequestLeaveViewModel.LeaveType.FORGOT_TO_CHECK_IN:
-                if (validateCheckinTime(request.getCheckinTime()) & validateProjectName(
-                        request.getProject())) {
+                if (validateCheckinTime(request.getCheckinTime())) {
                     return true;
                 }
                 break;
             case RequestLeaveViewModel.LeaveType.FORGOT_CARD_OUT:
             case RequestLeaveViewModel.LeaveType.FORGOT_TO_CHECK_OUT:
-                if (validateCheckoutTime(request.getCheckoutTime()) & validateProjectName(
-                        request.getProject())) {
+                if (validateCheckoutTime(request.getCheckoutTime())) {
                     return true;
                 }
                 break;
             case RequestLeaveViewModel.LeaveType.LEAVE_OUT:
-                if (validateCheckinTime(request.getCheckinTime())
-                        & validateCheckoutTime(request.getCheckoutTime())
-                        & validateCompensationFrom(request.getCompensation().getFromTime())
-                        & validateCompensationTo(request.getCompensation().getToTime())
-                        & validateReason(request.getReason())
-                        & validateProjectName(request.getProject())) {
+                if (validateCheckinTime(request.getCheckinTime()) & validateCheckoutTime(
+                        request.getCheckoutTime()) & validateCompensationFrom(
+                        request.getCompensation().getFromTime()) & validateCompensationTo(
+                        request.getCompensation().getToTime()) & validateReason(
```

Comment:
ERROR: [STYLE][NS_DANGEROUS_NON_SHORT_CIRCUIT] Potentially dangerous use of non-short-circuit logic in com.framgia.wsm.screen.requestleave.RequestLeavePresenter.validateDataInput(Request)
http://findbugs.sourceforge.net/bugDescriptions.html#NS_DANGEROUS_NON_SHORT_CIRCUIT